Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp

```diff
@@ -838,30 +838,32 @@ bool ResourceLoadStatisticsStore::tableExists(StringView tableName)
         ITP_RELEASE_LOG_DATABASE_ERROR("tableExists: failed to prepare statement");
         return false;
     }
-    if (scopedStatement->bindText(1, tableName) != SQLITE_OK) {
+    CheckedPtr checkedScopedStatement = scopedStatement.get();
+    if (checkedScopedStatement->bindText(1, tableName) != SQLITE_OK) {
         ITP_RELEASE_LOG_DATABASE_ERROR("tableExists: failed to bind parameter");
         return false;
     }
-    return scopedStatement->step() == SQLITE_ROW;
+    return checkedScopedStatement->step() == SQLITE_ROW;
 }
 
 void ResourceLoadStatisticsStore::deleteTable(StringView tableName)
 {
     ASSERT(tableExists(tableName));
     auto dropTableQuery = m_database.prepareStatementSlow(makeString("DROP TABLE "_s, tableName));
     ASSERT(dropTableQuery);
-    if (!dropTableQuery || dropTableQuery->step() != SQLITE_DONE)
+    if (!dropTableQuery || CheckedRef { *dropTableQuery }->step() != SQLITE_DONE)
         ITP_RELEASE_LOG_DATABASE_ERROR("deleteTable: failed to step statement");
 }
 
 bool ResourceLoadStatisticsStore::missingUniqueIndices()
 {
-    auto statement = m_database.prepareStatement("SELECT COUNT(*) FROM sqlite_master WHERE type = 'index'"_s);
-    if (!statement) {
+    auto preparedStatement = m_database.prepareStatement("SELECT COUNT(*) FROM sqlite_master WHERE type = 'index'"_s);
```

Comment:
Why can't we just use `CheckedPtr` here instead of checking nullity and using `CheckedRef`.